### ISSUE TITLE
abp `DefaultConventionalRegistrar` register instance logic

### ISSUE THREAD

**Issue by JadynWong** [abpframework/abp#2033]
The default ConventionalRegistrar `DefaultConventionalRegistrar` register instance to IOC.
When a service has multiple implementation interfaces, the default logic register a separate service instance for per interface.
https://github.com/abpframework/abp/blob/7ebb7a0097e8cc1f84c41078a802c596135e3dcf/framework/src/Volo.Abp.Core/Volo/Abp/DependencyInjection/DefaultConventionalRegistrar.cs#L27-L47

But I use autofac `RegisterService` injection serivce,it's logic register one single service instance for  per interface.

sample code
```c#
public interface ITestService
{
    string Say();
}

public interface IATestService
{
    string Say();
}

[ExposeServices(typeof(IATestService), typeof(ITestService), typeof(TestService))]
[Dependency(ServiceLifetime.Singleton)]
public class TestService : ITestService, IATestService, ISingletonDependency
{
    public Guid Id = Guid.NewGuid();
    public string Say()
    {
        return Id.ToString("D");
    }
}

public interface IAutofacTestService
{
    string Say();
}

public interface IAAutofacTestService
{
    string Say();
}

public class AutofacTestService : IAutofacTestService, IAAutofacTestService
{
    public Guid Id = Guid.NewGuid();
    public string Say()
    {
        return Id.ToString("D");
    }
}
```

```c#
var containerBuilder = context.Services.GetContainerBuilder();
            containerBuilder.RegisterType<AutofacTestService>().AsSelf().AsImplementedInterfaces().SingleInstance();
```

```c#
using (var abp = AbpApplicationFactory.Create<AbpConsoleModule>(options =>
{
    options.UseAutofac();
    //options.Services.AddConventionalRegistrar(new TestConventionalRegistrar());
}))
{
    abp.Initialize();

    Console.WriteLine($"TestService:         {abp.ServiceProvider.GetRequiredService<TestService>().Say()}");
    Console.WriteLine($"ITestService:        {abp.ServiceProvider.GetRequiredService<ITestService>().Say()}");
    Console.WriteLine($"IATestService:       {abp.ServiceProvider.GetRequiredService<IATestService>().Say()}");

    Console.WriteLine($"AutofacTestService:  {abp.ServiceProvider.GetRequiredService<AutofacTestService>().Say()}");
    Console.WriteLine($"IAutofacTestService: {abp.ServiceProvider.GetRequiredService<IAutofacTestService>().Say()}");
    Console.WriteLine($"IAAutofacTestService:{abp.ServiceProvider.GetRequiredService<IAAutofacTestService>().Say()}");

    abp.Shutdown();
}
```
the result.
![image](https://user-images.githubusercontent.com/13699353/67737136-11cf5a80-fa45-11e9-959d-e6e8578a4191.png)

I add a default TestConventionalRegistrar change `DefaultConventionalRegistrar` logic
```c#
var serviceTypes = ExposedServiceExplorer.GetExposedServices(type);

TriggerServiceExposing(services, type, serviceTypes);

Type firstRegisterType = null;

foreach (var serviceType in serviceTypes)
{
    ServiceDescriptor serviceDescriptor;
    if (firstRegisterType == null || dependencyAttribute?.ReplaceServices == true)
    {
        firstRegisterType = serviceType;
        serviceDescriptor = ServiceDescriptor.Describe(serviceType, type, lifeTime.Value);
    }
    else
    {
        serviceDescriptor = ServiceDescriptor.Describe(serviceType, provider => provider.GetRequiredService(firstRegisterType), lifeTime.Value);
    }

    if (dependencyAttribute?.ReplaceServices == true)
    {
        services.Replace(serviceDescriptor);
    }
    else if (dependencyAttribute?.TryRegister == true)
    {
        services.TryAdd(serviceDescriptor);
    }
    else
    {
        services.Add(serviceDescriptor);
    }
}
```
![image](https://user-images.githubusercontent.com/13699353/67737275-84d8d100-fa45-11e9-8f3c-5b045976430d.png)

The `DefaultConventionalRegistrar` inject logic for abp is best practices?

**Comment by JadynWong** [comment]
Because the default `DefaultConventionalRegistrar` exists.
The service will be register twice.
```c#
using (var abp = AbpApplicationFactory.Create<AbpConsoleModule>(options =>
{
    options.UseAutofac();
    options.Services.AddConventionalRegistrar(new TestConventionalRegistrar());
}))
{
    abp.Initialize();

    var TestServices = abp.ServiceProvider.GetRequiredService<IEnumerable<TestService>>();
    foreach (var item in TestServices)
    {
        Console.WriteLine($"TestService:         {item.Say()}");
    }
    var ITestServices = abp.ServiceProvider.GetRequiredService<IEnumerable<ITestService>>();
    foreach (var item in ITestServices)
    {
        Console.WriteLine($"ITestService:         {item.Say()}");
    }
    var IATestServices = abp.ServiceProvider.GetRequiredService<IEnumerable<IATestService>>();
    foreach (var item in IATestServices)
    {
        Console.WriteLine($"IATestService:         {item.Say()}");
    }
    Console.WriteLine($"AutofacTestService:  {abp.ServiceProvider.GetRequiredService<AutofacTestService>().Say()}");
    Console.WriteLine($"IAutofacTestService: {abp.ServiceProvider.GetRequiredService<IAutofacTestService>().Say()}");
    Console.WriteLine($"IAAutofacTestService:{abp.ServiceProvider.GetRequiredService<IAAutofacTestService>().Say()}");

    abp.Shutdown();
}
```
![image](https://user-images.githubusercontent.com/13699353/67737909-c4a0b800-fa47-11e9-8f4b-c4599c3c9eaa.png)

**Comment by JadynWong** [comment]
Any feedback? If this is wrong, please close it.Thanks.

**Comment by JadynWong** [comment]
I found that he was implemented in #3837.
